src/Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -37,6 +37,15 @@ public interface IBlockPuller : IDisposable
     {
         void Initialize(BlockPuller.OnBlockDownloadedCallback callback);
 
+        /// <summary>
+        /// Adds <see cref="NetworkPeerServices.NODE_WITNESS"/> to list of services that are required from all peers.
+        /// </summary>
+        /// <remarks>
+        /// In case some of the peers that we are already requesting block from don't support new
+        /// requirements those nodes will be released from their assignments.
+        /// </remarks>
+        void RequestWitnessOptions();
```

Comment:
This is problematic we are making witness part of the puller.
I think we can solve it differently bu passing the Network Enum here..

Reply:
It used to be part of the old puller before. 

Basically we need a way to filter out peers and pick only those that can provide us with the blocks we want.